src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseDataSourceServiceImpl.java

```diff
@@ -257,14 +248,14 @@ public DS findByUuidUnsecured(String uuid) throws NotExistException {
         return dataSourceRepository.suggest(query, studyId, pageRequest);
     }
 
-    @PreAuthorize("hasPermission(#uuid, 'DataSource', "
+    @PreAuthorize("hasPermission(#id, 'DataSource', "
```

Comment:
I guess, this is not the only use of uuid in PreAuthorize. What is more: what about permission evaluator itself? (and that crazy strategy for entity refreshing from db)

Reply:
it`s really single place where uuid has checked, others uses DataSource as object.
PermissionEvaluator still needs entity refreshing by uuid since uuid is used in analysis Files.